Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/CheckInWarningMatcher.kt

```diff
@@ -0,0 +1,27 @@
+package de.rki.coronawarnapp.presencetracing.risk
+
+import de.rki.coronawarnapp.eventregistration.checkins.CheckInRepository
+import de.rki.coronawarnapp.eventregistration.checkins.download.TraceTimeIntervalWarningRepository
+import de.rki.coronawarnapp.eventregistration.checkins.split.splitByMidnightUTC
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.flow.firstOrNull
+import javax.inject.Inject
+
+class CheckInWarningMatcher @Inject constructor(
+    private val checkInsRepository: CheckInRepository,
+    private val traceTimeIntervalWarningRepository: TraceTimeIntervalWarningRepository,
+    private val dispatcherProvider: DispatcherProvider
+) {
+    suspend fun execute(): List<CheckInWarningOverlap> {
+        val checkIns = checkInsRepository.allCheckIns.firstOrNull() ?: return emptyList()
+
+        val warningPackages = traceTimeIntervalWarningRepository.allWarningPackages.firstOrNull() ?: return emptyList()
```

Comment:
Some logging for the early aborts please :smile:

Reply:
I will add logging in the risk calculation PR as the logic here changes